packages/ragbits-core/src/ragbits/core/prompt/base.py

```diff
@@ -54,7 +54,7 @@ class BasePromptWithParser(Generic[OutputT], BasePrompt, metaclass=ABCMeta):
     """
 
     @abstractmethod
-    def parse_response(self, response: str) -> OutputT:
+    def parse_response(self, response: str | AsyncGenerator[str, None]) -> OutputT:
```

Comment:
Does it make sense to call `parse_response` for a streaming response?

Reply:
resolved by implementing separate method for streaming